design-decisions/2018-10-12-Condition-Split.md

```diff
@@ -0,0 +1,87 @@
+Splitting configurations upon new constraints on symbolic variables
+===================================================================
+
+Background
+----------
+
+When applying an axiom on a symbolic configuration new constraints may be 
+introduced on symbolic variables, coming from side conditions, or directly 
+from the unification process.
+
+### Example (from side conditions)
+
+Consider rewriting term `3 / A:Int` using 
+
+`rule X:Int / Y:Int => X /Int Y requires Y =/=Int 0`
+which would be written as 
+`Y =/= 0 = true /\ X:Int / Y:Int => X /Int Y`
+
+Then `3 / A:Int /\ (Y =/=Int 0 = true /\ X:Int / Y:Int)` will simplify to
+`(X:Int = 3 /\ Y:Int = A:Int) /\ (Y =/=Int 0) = true` 
```

Comment:
Not fully true, you forgot to add a `3 /Int A:Int` term in the simplification. And, since you're simplifying, the final result should actually be `3 /Int A:Int /\ A:Int =/= 0`.